### ISSUE TITLE
Returning promises (optionally)

### ISSUE THREAD

**Issue by diffcunha** [goto-bus-stop/split-require#18]
I'm building a PWA with Choo using asynchronous components and given that I'm using promises to deal with those it would be interesting if split-require could return a promise rather that using callbacks only. Something like:

```js
const splitRequire = require('split-require')

splitRequire('./SomeComponent').then(SomeComponent => { ... }) // promise
splitRequire('./SomeComponent', (err, SomeComponent) => { ... }) // callback
```

**Comment by goto-bus-stop** [comment]
yeah, this would be good--I think it might simplify the loader implementation a bit too, currently I'm jumping through a few hoops to combine callbacks but Promises already handle that. choo is also possibly going to be using Promises for async rendering soon™ so that would help with integration.

**Comment by goto-bus-stop** [comment]
Just remembered that Promises don't exist in IE11 which I would like to support, so I'll try to bolt them onto the old implementation instead

**Comment by goto-bus-stop** [comment]
finally released in :package: 3.1.0 :) thanks for the request!
